# --- ROCK-COPYRIGHT-NOTE-BEGIN ---
# 
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# Please add additional copyright information _after_ the line containing
# the ROCK-COPYRIGHT-NOTE-END tag. Otherwise it might get removed by
# the ./scripts/Create-CopyPatch script. Do not edit this copyright text!
# 
# ROCK Linux: rock-src/package/base/bison/gxx-attr-unused-fix.patch
# ROCK Linux is Copyright (C) 1998 - 2005 Clifford Wolf
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# 
# --- ROCK-COPYRIGHT-NOTE-END ---

diff -urN bison-1.875-old/data/yacc.c bison-1.875-new/data/yacc.c
--- bison-1.875-old/data/yacc.c	2002-12-28 09:36:02.000000000 +0100
+++ bison-1.875-new/data/yacc.c	2003-07-16 15:26:30.000000000 +0200
@@ -1114,8 +1114,10 @@
   /* Suppress GCC warning that yyerrlab1 is unused when no action
      invokes YYERROR.  */
 #if defined (__GNUC_MINOR__) && 2093 <= (__GNUC__ * 1000 + __GNUC_MINOR__)
+#if ! defined(__cplusplus)
   __attribute__ ((__unused__))
 #endif
+#endif
 
 ]b4_location_if([  yylerrsp = yylsp;
   *++yylerrsp = yyloc;])[
